Netty OKD Social Fix

isaacrivriv opened the pull request:
Added fix for okd social FAT given that Netty trims the Bearer token from it's trailing whitespace and so when parsed by the filter it accepts the token as is which shouldn't be


- [x] I have considered the risk of behavior change or other zero migration impact (https://github.com/OpenLiberty/open-liberty/wiki/Behavior-Changes).
- [ ] If this PR fixes an Issue, the description includes "Fixes #FILLMEIN" or "Resolves #FILLMEIN" (verify `release bug` label if applicable: https://github.com/OpenLiberty/open-liberty/wiki/Open-Liberty-Conventions).
- [ ] If this PR resolves an external Known Issue (including APARS), the description includes "Fixes #FILLMEIN" or "Resolves #FILLMEIN".

for #31988

isaacrivriv committed:
Added fix for okd social FAT given that Netty trims the Bearer token from it's trailing whitespace and so when parsed by the filter it accepts the token as is which shouldn't be

isaacrivriv commented:
!build ([view Open Liberty Personal Build](https://libh-proxy1.fyre.ibm.com/cognitive/pipelineAnalysis.html?pipelineId=88e3a994-ec54-488d-b4b7-54784545c4fe) - :white_check_mark: completed successfully!)

_Note: Target locations of links might be accessible only to IBM employees._

isaacrivriv committed:
Switched fix from a product perspective to a test perspective

isaacrivriv committed:
Updated copyright

isaacrivriv commented:
!build ([view Open Liberty Personal Build](https://libh-proxy1.fyre.ibm.com/cognitive/pipelineAnalysis.html?pipelineId=19b2cd46-4ff7-45a0-80f7-7ca65f049c7c) - :x: completed with errors/failures.)

_Note: Target locations of links might be accessible only to IBM employees._

isaacrivriv committed:
Added changes based off review feedback

jimmy1wu reviewed (commented):
hi isaac, i had something different in mind when changing the test expectation.

since right now with your change, we’d be testing `Authorzation: ""`.

and i think there’s still some value in testing `Authorization: Bearer`, so i was thinking to change:
https://github.com/OpenLiberty/open-liberty/blob/3efc8e44a70791f4ed542ac61652592fc500aa37/dev/com.ibm.ws.security.social_fat.okdServiceLogin/fat/src/com/ibm/ws/security/social/fat/okdServiceLogin/commonTests/OKDServiceLogin_GenericTests.java#L121
to expect the new error of a bad service account token:
https://github.com/OpenLiberty/open-liberty/blob/3efc8e44a70791f4ed542ac61652592fc500aa37/dev/com.ibm.ws.security.social_fat.okdServiceLogin/fat/src/com/ibm/ws/security/social/fat/okdServiceLogin/commonTests/OKDServiceLogin_GenericTests.java#L77

with a comment explaining netty trimming white space which causes the runtime to use `Bearer` as the access token due to not having a space after `Bearer`

you’d also have to update the test app to handle the access token being `Bearer` in this case:
https://github.com/OpenLiberty/open-liberty/blob/3efc8e44a70791f4ed542ac61652592fc500aa37/dev/com.ibm.ws.security.social_fat.okdServiceLogin/test-applications/StubbedOKDServiceLogin/src/com/ibm/ws/fat/OKDServiceLogin/UserValidationApi.java#L84-L103

i wouldn’t be opposed to adding a new test for `Authorization: ""` though in
https://github.com/OpenLiberty/open-liberty/blob/3efc8e44a70791f4ed542ac61652592fc500aa37/dev/com.ibm.ws.security.social_fat.okdServiceLogin/fat/src/com/ibm/ws/security/social/fat/okdServiceLogin/commonTests/OKDServiceLogin_GenericTests.java
by doing something like the `OKDServiceLogin_GenericTests_omitServiceAccountToken` test and using:
```
        updatedSocialTestSettings.setHeaderName(SocialConstants.BEARER_HEADER);
        updatedSocialTestSettings.setHeaderValue("");
 ```

jimmy1wu commented:
!build ([view Open Liberty Personal Build](https://libh-proxy1.fyre.ibm.com/cognitive/pipelineAnalysis.html?pipelineId=412d35b3-5cdd-4d6f-a2ef-84759ceda841) - :x: completed with errors/failures.)

_Note: Target locations of links might be accessible only to IBM employees._

jimmy1wu commented:
^ started a new build with the current change bc i'm kinda curious if it would affect other fat buckets, since the change is in common tooling

isaacrivriv committed:
Updated security tests
- Added new test to okd social with an empty authorization header
- Updated utility for returning a stripped bearer header
- Updated test to new exceptions with stripped bearer header

isaacrivriv commented:
!build ([view Open Liberty Personal Build](https://libh-proxy1.fyre.ibm.com/cognitive/pipelineAnalysis.html?pipelineId=e11beaa6-b23a-4977-8ad7-f289378d90fb) - :warning: completed with warnings.)

_Note: Target locations of links might be accessible only to IBM employees._
